File: platforms/web/examples/core/pass/add_depth_target.js
import { Renderer, Pass, Shader, Mesh } from "fragmentcolor";

const renderer = new Renderer();
const target = await renderer.createTextureTarget([64, 64]);

// One depth attachment shared across the 3D-content pass.
const depth = await renderer.createDepthTexture([64, 64]);

const mesh = new Mesh();
mesh.addVertex([0.0, 0.0, 0.0]);
mesh.addVertex([1.0, 0.0, 0.0]);
mesh.addVertex([0.0, 1.0, 0.0]);
mesh.addVertex([1.0, 1.0, 0.0]);
const shader = Shader.fromMesh(mesh);
const pass = new Pass("blobs"); pass.addShader(shader);

// Depth-test on — closer fragments win, the pass writes to the depth
// buffer so subsequent draws within the same pass see the depth.
pass.addDepthTarget(depth);

renderer.render(pass, target);

File: platforms/kotlin/examples/core/pass/add_depth_target.kt
import org.fragmentcolor.*

val renderer = Renderer()
val target = renderer.createTextureTarget(64u, 64u)

// One depth attachment shared across the 3D-content pass.
val depth = renderer.createDepthTexture(64u, 64u)

val mesh = Mesh()
mesh.addVertex(Vertex(listOf(0.0f, 0.0f, 0.0f)))
mesh.addVertex(Vertex(listOf(1.0f, 0.0f, 0.0f)))
mesh.addVertex(Vertex(listOf(0.0f, 1.0f, 0.0f)))
mesh.addVertex(Vertex(listOf(1.0f, 1.0f, 0.0f)))
val shader = Shader.fromMesh(mesh)
val pass = Pass("blobs"); pass.addShader(shader)

// Depth-test on — closer fragments win, the pass writes to the depth
// buffer so subsequent draws within the same pass see the depth.
pass.addDepthTarget(depth)

renderer.render(pass, target)